docs/memory_management.md

```diff
@@ -43,7 +43,7 @@ Shuffle threads have the highest priority to avoid priority inversion as the tas
 
 ### Deadlock Resolution
 
-Deadlocks occur when every active task has at least one thread blocked on memory allocation, either directly during its execution or indirectly by dependencies on shuffle blocked.
+Deadlocks occur when every active task has at least one dedicated thread blocked on memory allocation, and all of the pool threads working on that task are also blocked.
```

Comment:
Question: does the deadlock occur when both conditions are met that there is one or more blocked dedicated threads _and_ all pooled threads are blocked?